Tools/ardupilotwaf/cxx_checks.py

```diff
@@ -173,3 +173,11 @@ def check_libiio(cfg, env):
     if cfg.env['LIB_LIBIIO']:
         env.LIB += ['dl']
```

Comment:
Since we now have a check_libdl function it would be good if we use it here instead of just adding it.